### PR TITLE
More compat improvements

### PR DESCRIPTION
**Pull request by jezdez**
- don't trigger patcher on import
- don't trigger deprecationwarning in conda (memoize has been removed in conda trunk)
- prevents circular import on conda 22.9 (see #8)

### CODE REVIEW

**Review by mcg1969 — APPROVED** [review]
Looks good. I made the patch code more complex to prevent the circular import too, but if this can replace it, I'd prefer it.

**Review by mcg1969 — APPROVED** [review]

**Comment by mcg1969** [comment]
oops tests are failing, need to figure out why

**Comment by mcg1969** [comment]
Circular imports are still happening. The best way to fix this is to delay the patching further
```
Error loading anaconda_anon_usage: cannot import name 'diff_for_unlink_link_precs' from partially initialized module 'conda.core.solve' (most likely due to a circular import) (/Users/runner/work/anaconda-anon-usage/anaconda-anon-usage/testenv/lib/python3.11/site-packages/conda/core/solve.py)
```

**Comment by mcg1969** [comment]
For some reason `conda info` is swallowing the traceback even though I am trying to raise it.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -71,7 +71,11 @@ jobs:
         source $CONDA/etc/profile.d/conda.sh
         pkg=local::anaconda-anon-usage
         conda install anaconda-client constructor conda-index $pkg
-        conda create -p ./testenv $pkg conda==${{ matrix.cversion }} conda-libmamba-solver
+        if [[ ! "${{ matrix.cversion }}" =~ 4.* ]]; then
```
Comment (mcg1969) [inline .github/workflows/main.yml:74]: Don't try to install libmamba for conda 4.*, and make sure to skip the libmamba tests below

File: tests/test_config.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def _config(value):
 for ctype in ("env", "cfg"):
     if ctype == "cfg" and ENVKEY in os.environ:
         del os.environ[ENVKEY]
-    for mode in ("true", "false", "yes", "no", "on", "off", "default", ""):
+    for mode in ("true", "false", "yes", "no", "on", "off", "default"):
```
Comment (mcg1969) [inline tests/test_config.py:41]: There seem to be os-specific differences in behavior interpreting the blank string as the yaml config value. It's just not worth worrying about, as people are highly unlikely to use it

File: .github/workflows/main.yml (hunk 1 above)
Comment (jezdez) [reply .github/workflows/main.yml:74]: Good idea!

**Pull request merged**